src/navigator-card-body.jsx

```diff
@@ -0,0 +1,302 @@
+        return () => {
+            window.onresize = undefined;
+        };
+    });
+
+    useEffect(() => {
+        const onKeyboardNav = (e) => {
+            if (e.key === "ArrowRight") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx < sortedFiles.length - 1
+                        ? selectedIdx + 1
+                        : 0;
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowLeft") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx > 0
+                        ? selectedIdx - 1
+                        : sortedFiles.length - 1;
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowUp") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.max(selectedIdx - boxPerRow, 0);
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowDown") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.min(selectedIdx + boxPerRow, sortedFiles.length - 1);
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "Enter" && selected.length === 1) {
+                onDoubleClickNavigate(selected[0]);
+            }
+        };
+
+        if (!isMounted.current && !Dialogs.isActive()) {
+            isMounted.current = true;
+            document.addEventListener("keydown", onKeyboardNav);
+        }
+        if (Dialogs.isActive())
+            document.removeEventListener("keydown", onKeyboardNav);
+        return () => {
+            isMounted.current = false;
+            document.removeEventListener("keydown", onKeyboardNav);
+        };
+    }, [
+        setSelected,
+        sortedFiles,
+        boxPerRow,
+        selected,
+        onDoubleClickNavigate,
+        Dialogs
+    ]);
+
+    const resetSelected = e => {
+        if (e.target.id === "folder-view" || e.target.id === "navigator-card-body") {
+            setSelected([]);
+        }
+    };
+
+    const handleClick = (ev, file) => {
+        if (ev.detail > 1) {
+            onDoubleClickNavigate(file);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)